src/sentry/api/endpoints/organization_events.py

```diff
@@ -316,7 +316,14 @@ def get(self, request: Request, organization) -> Response:
             return Response(status=404)
 
         try:
-            snuba_params, params = self.get_snuba_dataclass(request, organization)
+            # If the organization allows joining and leaving projects, it means that their
+            # developers can visit issues from any projects. In the case of trace related issues,
+            # we need the endpoint to let us see events from all projects when this flag is enabled.
+            snuba_params, params = self.get_snuba_dataclass(
+                request,
+                organization,
+                check_global_views=not bool(organization.flags.allow_joinleave),
```

Comment:
This skips this block:
https://github.com/getsentry/sentry/blob/a190038cadc6f4deced6369fb36cfffe71694892/src/sentry/api/bases/organization_events.py#L123-L130